src/public/js/rs.js

```diff
@@ -45,13 +45,214 @@ var RS = (function() {
       }
     }
     if (errors) {
-      // TODO ERROR CORRECTION
-      throw new Error('Error correction not implemented yet');
+      // Perform B-M decoding
+      var syndromes = computeSyndromePoly(receivedMsg, n, k);
+      var bm = berlekampMassey(syndromes, n, k);
+      var errorLocations = chienSearch(bm.errorLocationPoly);
+      var errorMagnitudes = forneysFormula(errorLocations,
+                                           bm.errorMagnitudePoly);
+      var repairedMsg = repair(receivedMsg, errorLocations, errorMagnitudes);
```

Comment:
See below, consider just changing received message to be repaired. Since it makes sense in this content to actually repair the message.